services/inputhost/exthost.go

```diff
@@ -568,11 +584,16 @@ func (conn *extHost) sendMessage(pr *inPutMessage, extSendTimer *common.Timer, w
 		return
 	}
 
-	// If we reach the max sequence number, notify the extent controller but
-	// keep the pumps open
+	// just get the size of the actual data here
+	// ignoring other headers for the size computation
+	// Note: size of will not work here
+	conn.currSizeMB += float32(len(pr.putMsg.Data)) / bytesPerMB
```

Comment:
how about tracking the currSize and maxSize as bytes instead of MiBs ? That way, they can just be int64s and you also don't need the additional math conversion for every message.